# Dr. Dirac's Statistics Midterm
Grading a multiple choice exam is easy. But how much do multiple choice exams tell us about what a student really knows? Dr. Dirac is administering a statistics midterm exam and wants to use Bayes’ Theorem to help him understand the following:

- Given that a student answered a question correctly, what is the probability that she really knows the material?

Dr. Dirac knows the following probabilities based on many years of teaching:

- There is a question on the exam that 60% of students know the correct answer to.
- Given that a student knows the correct answer, there is still a 15% chance that the student picked the wrong answer.
- Given that a student does not know the answer, there is still a 20% chance that the student picks the correct answer by guessing.  

Using these probabilities, we can answer the question.

## 1
In order to use Bayes Theorem, we need to phrase our question as P(A|B).

What is A and B in this case?

In [1]:
# P(A|B) = what is the probability that she really knows the material Given that a student answered a question correctly
# P(A) = Prob she knows the material
# P(B) = Prob she gets the answer correct

## 2
What is the probability that the student knows the material?

In [2]:
prob_a_knows_material = 0.6

## 3
Given that the student knows the material, what is the probability that she answers correctly?

In [4]:
p_b_given_a_ans_correct_given_knows_material = 1 - 0.15
p_b_given_a_ans_correct_given_knows_material

0.85

## 4
What is the probability of any student answering correctly?

In [8]:
# taking into account correct knowing answer and guessing 
# takes into account the weighted average of knowing or not knowing the result

# knows answer and gets it correct
p_b_given_a_ans_correct_given_knows_material
# doesnt know the answer but gets it correct
p_doesnt_know_gets_correct = 0.2

# knows the correct answer
prob_a_knows_material
# doesn't know the answer
prob_doesnt_knows_material = 1 - prob_a_knows_material

prob_is_correct = (p_b_given_a_ans_correct_given_knows_material * prob_a_knows_material)\
                    + ((prob_doesnt_knows_material) * p_doesnt_know_gets_correct)
prob_is_correct

0.5900000000000001

## 5
Using the three probabilities and Bayes’ Theorem, calculate `P(knows material | answers correctly)`.

In [10]:
p_a_given_b = prob_a_knows_material * p_b_given_a_ans_correct_given_knows_material / prob_is_correct
p_a_given_b

0.8644067796610169

The probability of a student knowing the answer given they answered correctly is 86.44%.